re-organize install instructions

fmichonneau opened the pull request:
This is a draft to attempt to move forward the conversation about the organization of the installation instructions on the workshop template.

It implements the solution initially proposed in https://github.com/carpentries/maintainer-RFCs/issues/4 where the `curriculum` and `flavor` variables in the `_config.yml` file could be extended to accommodate combinations typically offered in SWC and LC workshops (in this PR only SWC workshops are being considered for now but if the solution is viable it could be extended to LC as well).

The installation instructions for each tool/software are moved into a separate sub-folder of `_include` which means that installation instructions could be shared across lesson programs (for instance, the git installation for SWC and LC workshops could be the same).

This file organization would make things easier for instructors setting up workshops website for the most typical cases and the simplified content of the `_include/<lesson program>/setup.html` makes it easier to toggle on or off manually which tool should appear on the workshop website.

While I'm generally in favor of having the installation instructions on each lesson, it creates an extra burden for learners to have to navigate to each of them to get the information they need.

@brownsarahm [pointed me](https://github.com/carpentries/lesson-infrastructure/issues/29) to [remote include](https://github.com/netrics/jekyll-remote-include). This Jekyll plugin would allow us to have a single copy of the installation instructions that could then be displayed on both the workshop website and in the lessons. However, we can't use this plugin with a Jekyll-based site on GitHub Pages. If the instructions live in the workshop-template repository, then we could use this plugin by relying on a continuous delivery system (for instance GitHub Actions) on the lesson repository. We would use a `jekyll build` command to generate an HTML version of the lesson site, the remote includes would be fetched at build time, and we could push the site to the `gh-pages` branch.

While the other way around (having the installation in the lesson repo, and using remote include on the workshop website) seems more appropriate, I don't think it would work because:

1. using GitHub Actions (or another CD system) would require the instructors to set up a Github Personal Authentication Tokens to allow the action to push to their repo after they import the repository within their own account to set up the workshop website. In itself, this isn't too difficult but it's an extra step, has potential security risk if the tokens are not handled correctly.
2. (while small) it introduces a greater chance of "spoofing" where a bad actor could replace the remote include target with deceiving code while creating a fake workshop website. While there is also a chance it could happen on the lessons, it's far less likely given that the maintainers would be able to review pull requests that would affect these files. 

I'd be curious to get the opinion of the @carpentries/lesson-infrastructure-committee and Lesson Maintainers on possible alternatives and path forward to address this.

References:

* see #660 
* see https://github.com/carpentries/maintainer-RFCs/issues/4

fmichonneau committed:
re-organize install instructions

zkamvar committed:
update online link to use page.address

The link to online workshops was pointing back to the initial site address, so I've changed it to link to the link to the online meeting.

zkamvar commented on _includes/swc/setup.html:
```diff
@@ -1,538 +1,27 @@
+{% include install_instructions/r.html %}
+{% else %}
+{% include warning-flavor.html %}
+{% endif %}
```
Should these install instructions be flipped?

```suggestion
{% if site.flavor == "r" %}
{% include install_instructions/r.html %}
{% elsif site.flavor == "python" %}
{% include install_instructions/python.html %}
{% else %}
{% include warning-flavor.html %}
{% endif %}
```

zkamvar reviewed (commented):
Overall, it looks good. The only concern I have is that it introduces a similar problem to #660, where the installation instructions default to being hidden on SWC sites.

What I would suggest instead, would be to issue a warning if the curriculum variable is not valid, but also include everything by default.

Removing the overall if statement and adding this at the top will work to accomplish that:

```liquid
{% comment %}
Since curriculm could be expanded, this will loop through all valid values.

Params:
  - `curricula`: an array of valid curricula. Liquid doesn't really know how to
    create an array on the spot, so we are left with this hack from SO:
    https://stackoverflow.com/q/22763180/2752888. A better way to do this would
    be to add it as an entry in the _config.yml so that it can easily be
    expanded in the future when there are more SWC lessons.
  - `invalid_curriculum`: indicator to display the warning
{% endcomment %}

{% assign curricula = "swc-gapminder|swc-inflammation"  | split: "|" %}
{% assign invalid_curriculum = true %}

{% for curriculum in curricula %}
  {% if curriculum == site.curriculum %}
    {% assign invalid_curriculum = false %}
    {% break %}
  {% endif %}
{% endfor %}

{% if invalid_curriculum %}

{% include warning-curriculum.html %}

{% endif %}

{% include install_instructions/shell.html %}
{% include install_instructions/git.html %}
{% include install_instructions/editor.html %}

...
```

zkamvar committed:
protect example jekyll variables

fmichonneau committed:
fix mix up

Co-authored-by: Zhian N. Kamvar <zkamvar@gmail.com>

ErinBecker commented:
I would particularly like to get feedback from @sstevens2 @munkm and @brownsarahm, as this proposed solution is intended to avoid problems like https://github.com/swcarpentry/git-novice/issues/735 and https://github.com/carpentries/workshop-template/issues/660. 🙂

ErinBecker commented:
@fmichonneau - this looks like a good solution to me.

brownsarahm commented on _config.yml:
```diff
@@ -16,20 +16,23 @@
+# swc-gapminder: for Software Carpentry based on the Gapminder dataset
 curriculum: "FIXME"
 
 # If there is both an R and python official version of a curriculum, change the
```
Does official here mean versions that are past alpha? if so, we should probably update this to be more specific.

fmichonneau replied:
Thanks Sarah, I'm adding links to the SWC and DC lesson page that lists our official lessons.

brownsarahm commented:
Overall I like it. I think if this was added to styles/carpentries theme it might be better for maintainance so that lessons could stop relying on the rendered copy of the template. 

Minor:  why are there both html and md copies of shell instructions?

fmichonneau committed:
remove unused file

fmichonneau commented:
> Minor: why are there both html and md copies of shell instructions?

lack of sleep :zzz:. I removed the extra file. Thank you for spotting this!

As for the other comments, we'll respond in the next few days after we have a chance to get more feedback.

sstevens2 committed:
Adding adblocker warning

fmichonneau commented:
@carpentries/lesson-infrastructure-committee we'd appreciate your comments and feedback on this pull request. Thank you!

sstevens2 commented:
I like the flexibility to have different install directions showing up and having the install tools outside of the lc/swc/dc structure in `_includes`.  
I'm not sure I completely understand the part about using the `remote include`.  If I understand correctly, the instructions would live in the template but then the lessons would use `remote include` to show the install directions in their setup page?

zkamvar committed:
update git for windows instructions

This is in reference to the changes that @uhilgert found in https://github.com/UA-Carpentries-Workshops/KEYS2020-Training/commit/3200a0e1dfca217c2f6cc7d2b5dd77399a1790eb

maxim-belkin commented on _includes/install_instructions/editor.html:
```diff
@@ -0,0 +1,66 @@
+
+
+<div id="editor"> {% comment %} Start of 'editor' section. {% endcomment %}
```
I think this comment is no longer necessary provided that the entire file is dedicated to 'editor' section.
Also, those two blank lines above stand out to me as not necessary :)

maxim-belkin commented on _includes/install_instructions/editor.html:
```diff
@@ -0,0 +1,66 @@
+        </p>
+        <p>
+          Others editors that you can use are
+          <a href="https://notepad-plus-plus.org/">Notepad++</a> or
```
it might be a good idea to make all links accessible by setting `alt` attributes. @ldko spent a fair bit of time working with pull requests related to this matter so if you need advice I'm sure she'll be able to provide one.

fmichonneau replied:
Hi @ldko, some guidance here would be great!

ldko replied:
hi @maxim-belkin @fmichonneau , what I had been working on was for `alt` attributes on `img` tags. I don't think `alt` text is expected on anchor links.

maxim-belkin replied:
My bad! Thanks, Lauren.

fmichonneau replied:
Thanks that also my understanding but wanted to be sure.

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,63 @@
+
+<div id="git"> {% comment %} Start of 'Git' section. GitHub browser compatibility
+  is given at https://help.github.com/articles/supported-browsers/{% endcomment %}
```
Same as with the editor.html -- let's remove the obvious 'Start of Git section' part or move/reword it to the top of the file. Not sure about GitHub browser compatibility bit -- if this is important, it shouldn't be in a comment form.

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,63 @@
+      <article role="tabpanel" class="tab-pane active" id="git-windows">
+        <p>
+          Git should be installed on your computer as part of your Bash
+          install (described above).
```
Here, "above" assumes that instructions from the `shell.html` precede the ones here. I'd suggest rephrasing to point to the `Shell` section. Add a note at the top of the document that instructions HERE rely on instructions in the Shell section.

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,63 @@
+      <article role="tabpanel" class="tab-pane active" id="git-macos">
+        <a href="https://www.youtube.com/watch?v=9LQhwETCdwY ">Video Tutorial</a>
+        <p>
+          <strong>For OS X 10.9 and higher</strong>, install Git for Mac
```
OS X -> macOS
"10.9 or higher" is too vague because 10.9 was released in 2013. I believe that in recent macOS versions one has to install Command Line Tools (the process for doing that differs on different versions of macOS)

fmichonneau replied:
I updated the OS X references but also now realizes that the recommended git osx installer hasn't been updated in over a year and I'm wondering if it's still the best recommendation for workshop participants :thinking:

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+    research computing, and great for general-purpose programming as
+    well.  Installing all of its research packages individually can be
+    a bit difficult, so we recommend
+    <a href="https://www.anaconda.com/distribution/">Anaconda</a>,
```
https://www.anaconda.com/products/individual

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+      <article role="tabpanel" class="tab-pane active" id="python-windows">
+        <a href="https://www.youtube.com/watch?v=xxQ0mzZ8UvA">Video Tutorial</a>
+        <ol>
+          <li>Open <a href="https://www.anaconda.com/distribution/#download-section">https://www.anaconda.com/distribution/#download-section</a> with your web browser.</li>
```
https://www.anaconda.com/products/individual#download-section

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+      <article role="tabpanel" class="tab-pane active" id="python-macos">
+        <a href="https://www.youtube.com/watch?v=TcSAln46u9U">Video Tutorial</a>
+        <ol>
+          <li>Open <a href="https://www.anaconda.com/distribution/#download-section">https://www.anaconda.com/distribution/#download-section</a> with your web browser.</li>
```
https://www.anaconda.com/products/individual#download-section

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="python-linux">
+        <ol>
+          <li>Open <a href="https://www.anaconda.com/distribution/#download-section">https://www.anaconda.com/distribution/#download-section</a> with your web browser.</li>
```
https://www.anaconda.com/products/individual#download-section

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+            file you just downloaded should appear.
+          </li>
+          <li>
+            Press <kbd>Enter</kbd>.
```
Return

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+            You will follow the text-only prompts.
+            To move through the text, press <kbd>Spacebar</kbd>.
+            Type <code>yes</code> and press enter to approve the license.
+            Press <kbd>Enter</kbd> to approve the default location
```
Return

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,98 @@
+            Type <code>yes</code> and press enter to approve the license.
+            Press <kbd>Enter</kbd> to approve the default location
+            for the files.
+            Type <code>yes</code> and press <kbd>Enter</kbd>
```
Return

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+      <article role="tabpanel" class="tab-pane active" id="shell-windows">
+        <a href="https://www.youtube.com/watch?v=339AEqk9c-8">Video Tutorial</a>
+        <ol>
+          <li>Download the Git for Windows <a href="https://git-for-windows.github.io/">installer</a>.</li>
```
https://gitforwindows.org/

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+          <li>
+            If your "HOME" environment variable is not set (or you don't know what this is):
+            <ol>
+              <li>Open command prompt (Open Start Menu then type <code>cmd</code> and press [Enter])</li>
```
`<kbd>Return</kbd>`

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+                Type the following line into the command prompt window exactly as shown:
+                <p><code>setx HOME "%USERPROFILE%"</code></p>
+              </li>
+              <li>Press [Enter], you should see <code>SUCCESS: Specified value was saved.</code></li>
```
`<kbd>Return</kbd>`

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+                <p><code>setx HOME "%USERPROFILE%"</code></p>
+              </li>
+              <li>Press [Enter], you should see <code>SUCCESS: Specified value was saved.</code></li>
+              <li>Quit command prompt by typing <code>exit</code> then pressing [Enter]</li>
```
`<kbd>Return</kbd>`

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+      <article role="tabpanel" class="tab-pane active" id="sql-windows">
+        <p>
+          <ul>
+            <li>Run git-bash from the start menu</li>
```
I believe Windows 8 doesn't have Start menu. I also think it's better to say `Git Bash` and `Start menu`.

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+        <p>
+          <ul>
+            <li>Run git-bash from the start menu</li>
+            <li>Copy the following <code>curl {{site.url}}{{site.baseurl}}/getsql.sh | bash</code></li>
```
`curl -fsSL` (for safety)

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+          <ul>
+            <li>Run git-bash from the start menu</li>
+            <li>Copy the following <code>curl {{site.url}}{{site.baseurl}}/getsql.sh | bash</code></li>
+            <li>Paste it into the window that git bash opened. If you're unsure, ask an instructor for help</li>
```
Git Bash

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+      <article role="tabpanel" class="tab-pane active" id="sql-web">
+        <p>
+          <ul>
+            <li>In case of problems: register for an account at <a href="http://pythonanywhere.com/">Python Anywhere</a></li>
```
https://www.pythonanywhere.com/

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+            <li>In case of problems: register for an account at <a href="http://pythonanywhere.com/">Python Anywhere</a></li>
+            <li>Download <a href="http://swcarpentry.github.io/sql-novice-survey/files/survey.db">survey.db</a></li>
+            <li>Click on files and upload survey.db</li>
+            <li>Click on dashboard and Choose new console <code>$ bash</code></li>
```
remove `$`

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+      <li role="presentation" class="active"><a data-os="windows" href="#sql-windows" aria-controls="Windows" role="tab" data-toggle="tab">Windows</a></li>
+      <li role="presentation"><a data-os="macos" href="#sql-macos" aria-controls="MacOS" role="tab" data-toggle="tab">MacOS</a></li>
+      <li role="presentation"><a data-os="linux" href="#sql-linux" aria-controls="Linux" role="tab" data-toggle="tab">Linux</a></li>
+      <li role="presentation"><a data-os="Web" href="#sql-web" aria-controls="Linux" role="tab" data-toggle="tab">Web</a></li>
```
data-os="web"  (lowercase w)?
aria-controls="Web"  (not Linux)?

maxim-belkin reviewed (changes requested):
I found a few issues but overall it looks good. I'd be happy to re-review once it's updated.

fmichonneau committed:
add link to role description

lachlandeer committed:
typo in online instruction location

fmichonneau committed:
fix formatting for flavor warning notice

fmichonneau committed:
add indication to check variables in curriculum warning

fmichonneau committed:
display install instructions even if SWC curriculum is invalid

fmichonneau committed:
incorporate @maxim-belkin comments

fmichonneau committed:
update documentation for setup instructions

fmichonneau committed:
use FIXME as default value for `flavor`

fmichonneau committed:
add table of contents in customization page

fmichonneau committed:
edit and formatting

fmichonneau commented:
Thank you all for your feedback.

I incorporated most of the suggested changes and updated the section in the "customization" page that explains how to change the installation instructions included in the index page (as a reminder a preview of this page is available at <https://deploy-preview-664--zen-ride-08ffd1.netlify.app/customization/index.html>).

This is now ready for another round of reviews.

We are planning to merge on June 22nd. If you'd like more time to review/comment, please let us know.

maxim-belkin commented on _includes/warning-curriculum.html:
```diff
@@ -1,4 +1,7 @@
-  Unrecognized value for variable <code>curriculum</code> set in <code>_config.yml</code>. Currently the variable is set to: <code>{{ site.curriculum }}</code>
+  Unrecognized value for variable <code>curriculum</code> set in
+  <code>_config.yml</code>. Currently the variable is set to:
+  <code>{{ site.curriculum }}</code>. Check that the values of
```
~that~

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+  <p><strong>If you installed Anaconda, it also has a copy of SQLite
+      <a href="https://github.com/ContinuumIO/anaconda-issues/issues/307">without support to <code>readline</code></a>.
+      Instructors will provide a workaround for it if needed.</strong></p>
+</div> {% comment %} End of 'SQLite' section. {% endcomment %}
```
I think we don't need these Liquid comments either.

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,58 @@
+
+<div id="sql"> {% comment %} Start of 'SQLite' section. {% endcomment %}
```
Same here: empty line and Liquid comment

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+      </article>
+    </div>
+  </div>
+</div> {% comment %} End of 'shell' section. {% endcomment %}
```
This last comment is not necessary.

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+<div id="shell"> {% comment %} Start of 'shell' section. {% endcomment %}
```
I think we don't need these `Start of ... section` / `End of ... section` comments

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,67 @@
+      </article>
+    </div>
+  </div>
+</div> {% comment %} End of 'Git' section. {% endcomment %}
```
~{% comment %} End of 'Git' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+<div id="openrefine"> {% comment %} Start of 'OpenRefine' section. {% endcomment %}
```
~{% comment %} Start of 'OpenRefine' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+      </article>
+    </div>
+  </div>
+</div> {% comment %} End of 'OpenRefine' section. {% endcomment %}
```
~ {% comment %} End of 'OpenRefine' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,101 @@
+<div id="python"> {% comment %} Start of 'Python' section. Remove the third paragraph if
```
I suggest moving this comment outside to the beginning of the file as you did in `git.html`

maxim-belkin commented on _includes/install_instructions/editor.html:
```diff
@@ -0,0 +1,64 @@
+      </article>
+    </div>
+  </div>
+</div> {% comment %} End of 'editor' section. {% endcomment %}
```
~ {% comment %} End of 'editor' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/r.html:
```diff
@@ -0,0 +1,55 @@
+      </article>
+    </div>
+  </div>
+</div> {% comment %} End of 'R' section. {% endcomment %}
```
~{% comment %} End of 'R' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,101 @@
+    which has instructions on how to test that everything was installed correctly.
+  </p>
+  {% endcomment %}
+</div> {% comment %} End of 'Python' section. {% endcomment %}
```
~{% comment %} End of 'Python' section. {% endcomment %}~

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,101 @@
+    (e.g., 3.6 is fine).
+  </p>
+
+  <p>
```
I suggest adding a Liquid comment here, something along the lines of:

> Please remove or comment out this paragraph using `<!--` and `-->` or `{% comment %}` and `{% endcomment %}` if you do not plan to use Jupyter Notebook environment.

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,101 @@
+    please go to <a href="setup/index.html">this page</a>,
+    which has instructions on how to test that everything was installed correctly.
+  </p>
+  {% endcomment %}
```
this commented out paragraph hides too deep to be noticeable. Is it supposed to appear at the end of ALL install instructions? Shall we move it (uncommented) to `_includes/*/setup.html`?

fmichonneau replied:
I'm going to delete this section because I don't think the scripts that are referred to on this page are currently useful.

fmichonneau committed:
remove unecessary comments and small edits

maxim-belkin commented on _extras/customization.md:
```diff
@@ -3,18 +3,26 @@ layout: page
+
+## Table of Content
+
+* TOC
```
Do we need this `* TOC` line?

maxim-belkin replied:
we do...

maxim-belkin commented on _extras/customization.md:
````diff
@@ -163,17 +224,25 @@ please follow the steps bellow:
-        $ git push origin gh-pages
+```bash
+$ git push origin gh-pages
+```
````
Shouldn't these be standard Carpentries code blocks?
```
~~~
...
~~~
{: .language-bash}
```

I also suggest adding `{% include links.md %}` to the end of this document.

fmichonneau replied:
:+1:

maxim-belkin commented on _includes/swc/setup.html:
```diff
@@ -1,538 +1,24 @@
 {% comment %}
-<div id="vm">
-  <h3>Virtual Machine</h3>
+These tools are not used very often, but you can move them outside this
```
Um... previously this comment was about "Virtual Machine" and now it's about install instructions for SQL and OpenRefine. I'd suggest removing "These tools are not used very often" and say instead that you have to uncomment this block if your workshop has SQL or OpenRefine sections.

fmichonneau replied:
The correspondence here between the section about VMs and the section about SQL/Openrefine is an artifact of the GitHub diff as they are both at the bottom of the file.

I removed the VM instructions as the image hasn't been updated in a very long time and I have heard of 
anyone using a VM to teach in a long time. 

I included "These tools are not used very often" as a way to make sure that new instructors wouldn't be tempted to include these instructions for workshops that don't use these tools. When @ErinBecker did a survey of the workshop pages recently, she's found that the setup instructions rarely matched what was actually taught. Am I remembering this correctly @ErinBecker?

maxim-belkin replied:
Ah, you mean topics. Got it.
Anyway, my point was about the wording on this comment. Perhaps, appending clarification like "in the Carpentries workshops" would be helpful:

> The following setup instructions are commented out because Carpentries workshops cover these topics less frequently. Please uncomment the lines that correspond to the topics covered in your workshop.

?

fmichonneau replied:
that's great, thanks Maxim!

maxim-belkin commented on _extras/design.md:
```diff
@@ -60,11 +60,11 @@ than Git itself.
-    (for example) `{{site.swc_site}}` in a page is replaced by the URL
-    of the main Software Carpentry web site.  Variables from the
+    (for example) `{% raw %}{{site.swc_site}}{% endraw %}` in a page is replaced by the URL
+    of the main Software Carpentry web site ({{site.swc_site}}).  Variables from the
```
I suggest adding `< >` so that we get a clickable link.
```
(<{{site.swc_site}}>)
```

munkm commented:
Hey, I just wanted to let you all know that I do plan to review this, I'm just behind on some other stuff. I'm hoping this coming week I'll have some time for a thorough review.

fmichonneau committed:
clarify that the tools aren't used often in carpentries workshops

fmichonneau committed:
use jekyll formatting for code blocks

maxim-belkin commented on _extras/customization.md:
```diff
@@ -3,18 +3,26 @@ layout: page
 * `carpentry` - to tell us which carpentry workshop this is, possible values are ("swc" for Software
   Carpentry workshops, "dc" for Data Carpentry workshops, "lc" for Library Carpentry Workshops, or
-  "cp" for general Carpentries events such as instructor trainings).
-* `curriculum` - for Data Carpentry, which one of the curriculum is being taught. Possible values
-  are: `dc-ecology`, `dc-genomics`, `dc-socsci`, `dc-geospatial`.
+  "cp" for general Carpentries events such as instructor trainings for which you should use
+  <https://github.com/carpentries/training-template> for your website template).
+* `curriculum` - for Data Carpentry and Software Carpentry, which one of the curriculum is being
+  taught. Possible values are: `dc-ecology`, `dc-genomics`, `dc-socsci`, `dc-geospatial`,
+  `swc-inflammation` and `swc-gapminder`.
```
```suggestion
* `carpentry` - to tell us which carpentry workshop this is. Possible values are:
    - `"swc"` for Software Carpentry workshops,
    - `"dc"` for Data Carpentry workshops,
    - `"lc"` for Library Carpentry workshops, and
    - `"cp"` for general Carpentries events such as instructor trainings (for which you should use
      <https://github.com/carpentries/training-template> as the website template)
* `curriculum` - to tell us which curriculum is being taught.
  At the moment, applicable to Software and Data Carpentry workshops only.
  Possible values are:
    - `"dc-ecology"`, `"dc-genomics"`, `"dc-socsci"`, and `"dc-geospatial"` for Data Carpentry workshops
    - `"swc-inflammation"` and `"swc-gapminder"` for Software Carpentry workshops
```

maxim-belkin replied:
I suggest using nested lists to show possible values. I also suggest adding double quotation marks around possible values to minimize possible confusion because things like `curriculum` and `carpentry` in `_config.yml` don't need those and possible values do.

maxim-belkin commented on _includes/swc/setup.html:
```diff
@@ -1,538 +1,25 @@
-<div id="vm">
-  <h3>Virtual Machine</h3>
+The following setup instructions are commented out because Carpentries workshops
+cover the these topics less frequently. Please uncomment the lines that
```
```suggestion
cover these topics less frequently. Please uncomment the lines that
```

maxim-belkin commented on _extras/customization.md:
```diff
@@ -3,18 +3,26 @@ layout: page
+* `curriculum` - for Data Carpentry and Software Carpentry, which one of the curriculum is being
+  taught. Possible values are: `dc-ecology`, `dc-genomics`, `dc-socsci`, `dc-geospatial`,
+  `swc-inflammation` and `swc-gapminder`.
 * `flavor` - `r` or `python` depending on which lessons are being taught at the workshop (currently
```
```suggestion
* `flavor` - `"r"` or `"python"` depending on which lessons are being taught at the workshop (currently
```

maxim-belkin reviewed (commented):
A few stylistic suggestions.

maxim-belkin commented on _extras/customization.md:
```diff
@@ -140,16 +148,69 @@ so that they show what you're actually planning to teach and when.  These
-software they don't need.
+If you need assistance with customizing the setup instructions for your website,
+feel free to ask your questions in the Carpentries
+[Instructors Slack channel](https://swcarpentry.slack.com/archives/C08BVNU00)
```
noticed `swcarpentry` here. If desired, Slack org name can be changed to `carpentries`:
https://slack.com/help/articles/201663443-Change-your-workspace-or-org-name-and-URL

fmichonneau replied:
yes, we have this in the works but there are some complications we need to resolve before we can finally move forward on this renaming.

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,67 @@
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="git-macos">
+        <a href="https://www.youtube.com/watch?v=9LQhwETCdwY ">Video Tutorial</a>
```
we can embed these videos....

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+        <p>Unzip the downloaded file into the OpenRefine directory by double-clicking it.</p>
+        <p>Go to your newly created OpenRefine directory.</p>
+        <p>Launch OpenRefine by dragging the icon into the Applications folder.</p>
+        <p>Use <code>Ctrl-click/Open ... </code> to launch it.</p>
```
```suggestion
<p>Use <kbd>Ctrl</kbd>-click/Open to launch it.</p>
```

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+          It will not run correctly in Internet Explorer.
+        </p>
+        <p>Download software from <a href="http://openrefine.org/">http://openrefine.org/</a></p>
+        <p>Create a new directory called OpenRefine.</p>
```
All of these `<p>`s should be changed to `<li>`. And `<ol>.... </ol>` added around them.

maxim-belkin commented on _includes/warning-curriculum.html:
```diff
@@ -1,4 +1,7 @@
+  Unrecognized value for variable <code>curriculum</code> set in
+  <code>_config.yml</code>. Currently the variable is set to:
+  <code>{{ site.curriculum }}</code>. Check the values of
+  <code>carpentry</code> and <code>curriculum</code>.
```
```suggestion
  <code>carpentry</code> and <code>curriculum</code> in the <code>_config.yml</code> file
located in the root folder of this workshop website.
```

maxim-belkin commented on _includes/warning-flavor.html:
```diff
@@ -1,3 +1,7 @@
+  <div class="alert alert-warning">
+    <p>
+      Unrecognized value for variable <code>flavor</code> set in
+      <code>_config.yml</code>. Valid values are <code>r</code> and
+      <code>python</code>, currently set to <code>{{ site.flavor }}</code>.
```
```suggestion
      <code>_config.yml</code>. Valid values are <code>"r"</code> and
      <code>"python"</code>, currently set to <code>{{ site.flavor }}</code>.
```

fmichonneau committed:
add warning for invalid SWC curricula

maxim-belkin commented on index.md:
```diff
@@ -65,6 +65,18 @@ It looks like you are setting up a website for a Data Carpentry curriculum but y
+</div>
+{% endunless %}
+{% endif %}
+
```
After this change, we will see the above warning on this website.

zkamvar replied:
For what it's worth, we currently already see [an alert on the site](https://github.com/carpentries/workshop-template/pull/664/commits/592b0d0af149bc7d2c214893ebc67532ff2fdea6#diff-d680e8a854a7cbad6d490c445cba2ebaL42-L48), so another alert below it will not be too detrimental to the current status quo of the site.

The question becomes: who is the audience for https://carpentries.github.io/workshop-template? I know at the moment, several lessons point to the install page for updated installation instructions, but am I wrong in thinking that learners would not be visiting this site otherwise? Is there a better way to fail fast, early, and obviously so that the facilitators can set up the website properly?

maxim-belkin replied:
We can hide divs that belong to class alert and alert-danger using jQuery:
```js
<script src="https://code.jquery.com/jquery-3.5.0.js"></script>
<script>
$( document ).ready(function() {
  if (location.href.startsWith("https://carpentries.github.io/workshop-template/")) {
      $("div.alert.alert-danger").hide();
  }
});
</script>
```

zkamvar replied:
> We can hide divs that belong to class alert and alert-danger using jQuery

If we didn't want the alerts to appear on the page, we could just not place them there in the first place.

To determine if we should have alerts like this on the page, we have to ask: who is the target audience and how does it help us to have the alerts?

maxim-belkin replied:
The alerts have to appear on the workshop (event) page if preparer forgot to update the fields but they don't have to appear on the workshop-template's website.

maxim-belkin commented on _includes/install_instructions/editor.html:
```diff
@@ -0,0 +1,64 @@
+          Please ask your instructor to help you do this.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="editor-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="editor-macos">
```

maxim-belkin commented on _includes/install_instructions/editor.html:
```diff
@@ -0,0 +1,64 @@
+          <a href="https://www.sublimetext.com/">Sublime Text</a>.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="editor-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="editor-linux">
```

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,67 @@
+          <a href="#shell">Shell installation instructions</a>).
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="git-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="git-macos">
```

maxim-belkin commented on _includes/install_instructions/git.html:
```diff
@@ -0,0 +1,67 @@
+          <a href="http://sourceforge.net/projects/git-osx-installer/files/">available here</a>.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="git-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="git-linux">
```

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+        <p>Launch OpenRefine by clicking <code>openrefine.exe</code> (this will launch a command prompt window, but you can ignore that - just wait for OpenRefine to open in the browser).</p>
+        <p>If you are using a different browser, or if OpenRefine does not automatically open for you, point your browser at <a href="http://127.0.0.1:3333/">http://127.0.0.1:3333/</a> or <a href="http://localhost:3333">http://localhost:3333</a> to use the program.</p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="openrefine-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="openrefine-macos">
```

maxim-belkin commented on _includes/install_instructions/openrefine.html:
```diff
@@ -0,0 +1,51 @@
+        <p>Use <code>Ctrl-click/Open ... </code> to launch it.</p>
+        <p>If you are using a different browser, or if OpenRefine does not automatically open for you, point your browser at <a href="http://127.0.0.1:3333/">http://127.0.0.1:3333/</a> or <a href="http://localhost:3333">http://localhost:3333</a> to use the program.</p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="openrefine-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="openrefine-linux">
```

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,103 @@
+          <li>Install Python 3 by running the Anaconda Installer, using all of the defaults for installation <em>except</em> make sure to check <strong>Add Anaconda to my PATH environment variable</strong>.</li>
+        </ol>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="python-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="python-macos">
```

maxim-belkin commented on _includes/install_instructions/python.html:
```diff
@@ -0,0 +1,103 @@
+          <li>Install Python 3 by running the Anaconda Installer using all of the defaults for installation.</li>
+        </ol>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="python-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="python-linux">
```

maxim-belkin commented on _includes/install_instructions/r.html:
```diff
@@ -0,0 +1,55 @@
+          for example when installing R packages.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="rstats-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="rstats-macos">
```

maxim-belkin commented on _includes/install_instructions/r.html:
```diff
@@ -0,0 +1,55 @@
+          <a href="https://www.rstudio.com/products/rstudio/download/#download">RStudio IDE</a>.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="rstats-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="rstats-linux">
```

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+        </ol>
+        <p>This will provide you with both Git and Bash in the Git Bash program.</p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="shell-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="shell-macos">
```

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+          this Apple Support article</a> and follow the instructions on "How to change your default shell".
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="shell-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="shell-linux">
```

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="sql-macos">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="sql-macos">
```

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+          SQLite comes pre-installed on macOS.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="sql-linux">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="sql-linux">
```

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+          SQLite comes pre-installed on Linux.
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane active" id="sql-web">
```
```suggestion
      <article role="tabpanel" class="tab-pane" id="sql-web">
```

maxim-belkin reviewed (changes requested):
class "active" is added to articles when user clicks on corresponding `<li>` element. We shouldn't be adding them to all articles by default. This is not observed here, but if instead of "include"-ing these HTML files we were to add their contents, we'd see contents of all "active" articles. In the suggestions below, I kept class "active" only in the first article in a file.

ErinBecker commented:
> Hey, I just wanted to let you all know that I do plan to review this, I'm just behind on some other stuff. I'm hoping this coming week I'll have some time for a thorough review.

@munkm - Thanks so much for offering to review. @fmichonneau is working on incorporating the detailed feedback from @maxim-belkin (thank you Maxim!) and we're looking at finalizing this on Friday. Do you think you'd be able to look this over by the end of the week?

maxim-belkin commented on _includes/install_instructions/shell.html:
```diff
@@ -0,0 +1,123 @@
+<div id="shell">
+  <h3>The Bash Shell</h3>
+  <p>
+    Bash is a commonly-used shell that gives you the power to do simple
```
remove "simple"

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+
+  <p>
+    SQL is a specialized programming language used with databases.  We
+    use a simple database manager called
```
remove "simple"

maxim-belkin commented on index.md:
```diff
@@ -74,6 +86,8 @@ are not using Eventbrite, or leave it in, since it will not be
 {% endcomment %}
 {% if page.eventbrite %}
+<strong>Some adblockers block the registration window. If you do not see the 
+  registration box below, please check your adblocker settings.</strong>
```
We might be able to detect this with jQuery. 

```js
$(document).ready(function(){
  if ($('iframe').height() == 0) {
  $('iframe').before('<p>Looks like your adblocker blocked registration window. Please navigate to <a href="https://www.eventbrite.com/tickets-external?eid={{page.eventbrite}}&ref=etckt">https://www.eventbrite.com/tickets-external?eid={{page.eventbrite}}&ref=etckt</a> to register.</p>');
  }
});
```

maxim-belkin reviewed (changes requested):
A few more suggestions / comments.

fmichonneau committed:
remove active class from tabpanels

Co-authored-by: Maxim Belkin <maxim.belkin@gmail.com>

fmichonneau committed:
remove unused/unecessary GA

fmichonneau committed:
consolidate javascript scripts in a single file

fmichonneau committed:
add scripts to hide warning, inform eventbrite missing

fmichonneau committed:
bring changes

maxim-belkin committed:
Fix customizations

maxim-belkin commented on _extras/customization.md:
```diff
@@ -163,17 +230,32 @@ please follow the steps bellow:
+~~~
+{: .language-bash}
+
+
```
Currently, this list appears as
```
1. ...

1. ...

1. ...
```

I submitted #674 against this branch to fix it.

maxim-belkin commented on _extras/customization.md:
```diff
@@ -3,18 +3,32 @@ layout: page
+    - `"lc"` for Library Carpentry workshops, and
+    - `"cp"` for general Carpentries events such as instructor trainings (for which you should use
+      <https://github.com/carpentries/training-template> as the website template).
+
```
it doesn't break anything but I don't think we need this empty line.

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+          </ul>
+
+          <p>If you want to do this manually, download <a href="https://www.sqlite.org/2019/sqlite-tools-win32-x86-3270200.zip">sqlite3</a>, make a bin directory in the user's home directory, unzip sqlite3, move it into the bin directory, and then add the bin directory to the path.</p>
+
+        </p>
```
```suggestion


```

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+        </p>
+      </article>
+      <article role="tabpanel" class="tab-pane" id="sql-web">
+        <p>
+          <ul>
+            <li>In case of problems: register for an account at <a href="https://www.pythonanywhere.com/">https://www.pythonanywhere.com</a></li>
+            <li>Download <a href="http://swcarpentry.github.io/sql-novice-survey/files/survey.db">survey.db</a></li>
+            <li>Click on files and upload survey.db</li>
+            <li>Click on dashboard and Choose new console <code>bash</code></li>
+            </ul>
+        </p>
```
```suggestion
        <ul>
          <li>In case of problems: register for an account at <a href="https://www.pythonanywhere.com/">https://www.pythonanywhere.com</a></li>
          <li>Download <a href="http://swcarpentry.github.io/sql-novice-survey/files/survey.db">survey.db</a></li>
          <li>Click on files and upload survey.db</li>
          <li>Click on dashboard and Choose new console <code>bash</code></li>
        </ul>
```

maxim-belkin commented on _includes/install_instructions/sql.html:
```diff
@@ -0,0 +1,57 @@
+    </ul>
+    <div class="tab-content">
+      <article role="tabpanel" class="tab-pane active" id="sql-windows">
+        <p>
```
```suggestion
```

maxim-belkin reviewed (commented):
Looks great! Minor HTML clean-ups.

fmichonneau committed:
remove empty line

fmichonneau committed:
remove dangling missing tag

fmichonneau committed:
Merge branch 'gh-pages' into install-instructions

fmichonneau commented:
Thank you for your comments and review! especially Maxim for spotting all the small but important details.
@munkm if you have comments about these changes, feel free to open a new issue/pull request or add more comments here.
